tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_PrimitiveGhostData.cpp

```diff
@@ -50,6 +54,45 @@ void test_subcells(
         CHECK_ITERABLE_APPROX(get<tag>(recons_prims), get<tag>(prims));
       });
 }
+
+template <size_t Dim>
+void test_dg(const gsl::not_null<std::mt19937*> gen,
+             const gsl::not_null<std::uniform_real_distribution<>*> dist) {
+  using MassDensity = NewtonianEuler::Tags::MassDensity<DataVector>;
+  using Velocity = NewtonianEuler::Tags::Velocity<DataVector, Dim>;
+  using SpecificInternalEnergy =
+      NewtonianEuler::Tags::SpecificInternalEnergy<DataVector>;
+  using Pressure = NewtonianEuler::Tags::Pressure<DataVector>;
+  using prim_tags =
+      tmpl::list<MassDensity, Velocity, SpecificInternalEnergy, Pressure>;
+  using PrimVars = Variables<prim_tags>;
+  using prims_to_reconstruct_tags = tmpl::list<MassDensity, Velocity, Pressure>;
+  using ReconsPrimVars = Variables<prims_to_reconstruct_tags>;
+
+  const Mesh<Dim> dg_mesh{5, Spectral::Basis::Legendre,
+                          Spectral::Quadrature::GaussLobatto};
+  const Mesh<Dim> subcell_mesh = evolution::dg::subcell::fd::mesh(dg_mesh);
+
+  // NOLINTNEXTLINE(modernize-use-auto)
+  const PrimVars prims = make_with_random_values<PrimVars>(
+      gen, dist, dg_mesh.number_of_grid_points());
+
+  auto box = db::create<
+      db::AddSimpleTags<::Tags::Variables<prim_tags>, domain::Tags::Mesh<Dim>,
+                        evolution::dg::subcell::Tags::Mesh<Dim>>>(
+      prims, dg_mesh, subcell_mesh);
+  const ReconsPrimVars recons_prims =
+      db::mutate_apply<NewtonianEuler::subcell::PrimitiveGhostDataToSlice<Dim>>(
+          make_not_null(&box));
+
+  const PrimVars subcell_prims = evolution::dg::subcell::fd::project(
+      prims, dg_mesh, subcell_mesh.extents());
+  tmpl::for_each<prims_to_reconstruct_tags>(
+      [&recons_prims, &subcell_prims](auto tag_v) {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        CHECK_ITERABLE_APPROX(get<tag>(recons_prims), get<tag>(subcell_prims));
```

Comment:
I don't know what's compared to what... can one of these be prefixed with `expected_`?

As in the other commit, would `recons_prims` -> `prims_to_reconstruct` be accurate? if so, I'd find it easier to put in context.